File: zad2.py
def algorithm(c, m): # Алгоритм нахождения НОД(c, m) и числа d 
    
    one = [m, 0]
    two = [c, 1]
    three = [0, 0]
    q = 0
    while (two[0] != 0):
        q = one[0] // two[0]
        three[0] = one[0] % two[0]
        three[1] = one[1] - (q * two[1])
        one[0], one[1] = two[0], two[1]
        two[0], two[1] = three[0], three[1]
        
    return one[0], one[1]


print("Нахождение d, чтобы c*d mod m = 1:")

c = int(input("Введите число c: "))
m = int(input("Введите число m: "))

gcd, d = algorithm(c, m)

if (gcd != 1):
    print("Такого d не существует, т.к. c и m не взаимнопростые.")
else:
    if (d < 0):
        d += m
    print("Число d:", d)
    print(c*d, "mod", m, "=", (c*d) % m)

File: zad2.cpp
#include <iostream>
#include <vector>

using namespace std;

void algorithm(int c, int m, int &gcd, int &d) { // Алгоритм нахождения НОД(c, m) и числа d 

    int q; 
    vector<int> one = {m, 0};
    vector<int> two = {c, 1};
    vector<int> three(2);
    while (two[0] != 0) {
        q = one[0] / two[0];
        three[0] = one[0] % two[0];
        three[1] = one[1] - (q * two[1]);
        one[0] = two[0]; one[1] = two[1];
        two[0] = three[0]; two[1] = three[1];
    }

    gcd = one[0], d = one[1];
}

int main(){
    system ("chcp 65001");

    cout << "Нахождение d, чтобы c*d mod m = 1:" << endl;
   
    int c, m;
    cout << "Введите числа c и m: ";
    cin >> c >> m;

    int gcd, d;
    algorithm(c, m, gcd, d);
    
    if (gcd != 1) {
        cout << "Такого d не существует, т.к. c и m не взаимнопростые.";
    } else {
        if (d < 0) {
            d += m;
        }
        cout << "Число d: " << d << endl << c*d << " mod " << m << " = " << (c*d) % m;
    }

    return 0;
}
